lib/tests/unit/connector/mock_server.cc

```diff
@@ -0,0 +1,48 @@
+#include "mock_server.hpp"
+
+#include <cpp-pcp-client/util/thread.hpp>
+#include <cpp-pcp-client/util/chrono.hpp>
+
+namespace PCPClient {
+
+MockServer::MockServer(int timeout,
+                       uint16_t port,
+                       std::string certPath,
+                       std::string keyPath)
+    : websocketpp::server<websocketpp::config::asio_tls>(), timeout_(timeout), certPath_(std::move(certPath)), keyPath_(std::move(keyPath))
+{
+    namespace asio = websocketpp::lib::asio;
+    init_asio();
+    set_tls_init_handler([&](websocketpp::connection_hdl hdl) {
+        auto ctx = websocketpp::lib::make_shared<asio::ssl::context>(asio::ssl::context::tlsv1);
+        ctx->set_options(asio::ssl::context::default_workarounds |
+                         asio::ssl::context::no_sslv2 |
+                         asio::ssl::context::no_sslv3 |
+                         asio::ssl::context::single_dh_use);
+        ctx->use_certificate_file(certPath_, asio::ssl::context::file_format::pem);
+        ctx->use_private_key_file(keyPath_, asio::ssl::context::file_format::pem);
+        return ctx;
+    });
+    listen(port);
+}
+
+MockServer::~MockServer()
+{
+    stop();
+    Util::this_thread::sleep_for(Util::chrono::milliseconds(2*timeout_));
+}
+
+void MockServer::go()
+{
+    start_accept();
+    boost::thread bt([&]() { run(); });
```

Comment:
At first, I thought that calling run() in a separate thread and returning right away could cause a race, as stated by [websocketpp](http://www.zaphoyd.com/websocketpp/manual/building-program-websocket), connections are accepted when the underlying asio::io_service is effectively running.
But it doesn't seem to be the case.

By the way, I couldn't find any io_service inspector to check whether or not it runs. Only stopped(), which is exposed by websocketpp's endpoint.